In [2]:
import pandas as pd

url = "https://raw.githubusercontent.com/stedy/Machine-Learning-with-R-datasets/master/insurance.csv"

df = pd.read_csv(url)

df.head()

,age,sex,bmi,children,smoker,region,charges
0,19,female,27.900,0,yes,southwest,16884.92400
1,18,male,33.770,1,no,southeast,1725.55230
2,28,male,33.000,3,no,southeast,4449.46200
3,33,male,22.705,0,no,northwest,21984.47061
4,32,male,28.880,0,no,northwest,3866.85520


In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


In [4]:
import sqlite3

conn = sqlite3.connect("healthcare.db")
df.to_sql("insurance", conn, if_exists="replace", index=False)

1338

In [5]:
query = """
SELECT region, ROUND(AVG(charges),2) AS avg_cost
FROM insurance
GROUP BY region
ORDER BY avg_cost DESC;
"""

import pandas as pd
pd.read_sql(query, conn)

,region,avg_cost
0,southeast,14735.41
1,northeast,13406.38
2,northwest,12417.58
3,southwest,12346.94


In [6]:
query = """
SELECT smoker, ROUND(AVG(charges),2) AS avg_cost
FROM insurance
GROUP BY smoker;
"""

pd.read_sql(query, conn)

,smoker,avg_cost
0,no,8434.27
1,yes,32050.23


Key Insights

1. The Southeast region has the highest average medical charges among all regions.
2. Smoking status is the strongest cost driver in the dataset.
3. Smokers incur average healthcare costs of 32,050.23, compared with 8,434.27 for non-smokers.
4. On average, smokers cost nearly four times more than non-smokers.

In [7]:
query = """
SELECT
  CASE
    WHEN bmi < 18.5 THEN 'Underweight'
    WHEN bmi BETWEEN 18.5 AND 24.9 THEN 'Normal'
    WHEN bmi BETWEEN 25 AND 29.9 THEN 'Overweight'
    ELSE 'Obese'
  END AS bmi_category,
  ROUND(AVG(charges),2) AS avg_cost
FROM insurance
GROUP BY bmi_category
ORDER BY avg_cost DESC;
"""

pd.read_sql(query, conn)

,bmi_category,avg_cost
0,Obese,15479.55
1,Overweight,10993.99
2,Normal,10379.50
3,Underweight,8852.20


In [8]:
query = """
SELECT
  CASE
    WHEN age < 30 THEN 'Young'
    WHEN age BETWEEN 30 AND 50 THEN 'Middle Age'
    ELSE 'Senior'
  END AS age_group,
  ROUND(AVG(charges),2) AS avg_cost
FROM insurance
GROUP BY age_group
ORDER BY avg_cost DESC;
"""

pd.read_sql(query, conn)

,age_group,avg_cost
0,Senior,18084.99
1,Middle Age,13253.93
2,Young,9182.49


In [9]:
query = """
SELECT age, sex, bmi, children, smoker, region, charges
FROM insurance
ORDER BY charges DESC
LIMIT 10;
"""

pd.read_sql(query, conn)

,age,sex,bmi,children,smoker,region,charges
0,54,female,47.410,0,yes,southeast,63770.42801
1,45,male,30.360,0,yes,southeast,62592.87309
2,52,male,34.485,3,yes,northwest,60021.39897
3,31,female,38.095,1,yes,northeast,58571.07448
4,33,female,35.530,0,yes,northwest,55135.40209
5,60,male,32.800,0,yes,southwest,52590.82939
6,28,male,36.400,1,yes,southwest,51194.55914
7,64,male,36.960,2,yes,southeast,49577.66240
8,59,male,41.140,1,yes,southeast,48970.24760
9,44,female,38.060,0,yes,southeast,48885.13561


Final Insights

1. Smoking is the strongest predictor of high healthcare costs, with smokers incurring nearly four times higher average charges than non-smokers.

2. Obesity significantly increases healthcare expenses, with obese individuals having the highest average cost among all BMI categories.

3. Healthcare costs increase consistently with age, with senior individuals incurring the highest expenses.

4. The most expensive patients are predominantly smokers with high BMI, indicating a compounding effect of risk factors.

5. Regional differences exist, with the Southeast showing the highest average costs, although the variation is less significant compared to smoking and BMI.

Business Recommendations

1. Invest in smoking cessation programs to reduce high-cost patient segments.

2. Promote preventive health initiatives targeting obesity and high BMI individuals.

3. Focus early intervention strategies on middle-aged customers before they transition into higher-cost senior groups.

4. Develop targeted healthcare plans for high-risk segments combining age, smoking status, and BMI.

5. Allocate healthcare resources more efficiently in high-cost regions such as the Southeast.